src/main/java/org/jenkinsci/plugins/junitrealtimetestreporter/RealtimeTestResultAction.java

```diff
@@ -59,105 +50,26 @@
-
-    public RealtimeTestResultAction(final AbstractBuild<?, ?> owner) {
-
-        super(owner);
-    }
-
-    @Override
-    public TestResult getResult() {
-
-        // Refresh every 1/100 of a job estimated duration but not more often than every 5 seconds
-        final long thrashold = Math.max(5000, owner.getEstimatedDuration() / 100);
-        if (updated > System.currentTimeMillis() - thrashold) {
-            LOGGER.fine("Cache hit");
-            return result;
-        }
-
-        result = parse();
-        updated = System.currentTimeMillis();
-        return result;
-    }
-
-    @Override
-    public int getFailCount() {
-
-        if (getResult() == null) return 0;
-        return getResult().getFailCount();
-    }
-
-    @Override
-    public int getTotalCount() {
-
-        if (getResult() == null) return 0;
-        return getResult().getTotalCount();
-    }
-
-    public TestResult getTarget() {
-
-        if (!owner.isBuilding()) {
-            LOGGER.warning("Dangling RealtimeTestResultAction on " + owner + ". Probably not finalized correctly.");
-            detachFrom(owner);
-            throw new HttpRedirect(owner.getUrl());
-        }
-
-        if (getResult() != null) return getResult();
-
-        return new NullTestResult(this);
-    }
+    public RealtimeTestResultAction() {}
 
     @Override
     public String getDisplayName() {
-
         return "Realtime Test Result";
     }
 
     @Override
     public String getUrlName() {
-
         return "realtimeTestReport";
     }
 
     @Override
-    public String getIconFileName() {
-
-        return super.getIconFileName();
-    }
-
-    private TestResult parse() {
-
+    protected TestResult parse() throws IOException, InterruptedException {
         final JUnitResultArchiver archiver = getArchiver(this.owner);
-
-        try {
-
-            final long started = System.currentTimeMillis();
-            final TestResult result = new JUnitParser(archiver.isKeepLongStdio())
-                    .parse(getGlob(archiver), this.owner, null, null)
-            ;
-            LOGGER.log(Level.INFO, "Parsing took {0} ms", System.currentTimeMillis() - started);
-
-            result.setParentAction(this);
-            return result;
-        } catch (AbortException ex) {
-            // Thrown when there are no reports or no workspace witch is normal
-            // at the beginning the build. This is also a signal that there are
-            // no reports to update (already parsed was excluded and no new have
-            // arrived so far).
-            LOGGER.fine("No new reports found.");
-        } catch (InterruptedException ex) {
-            LOGGER.log(Level.WARNING, "Unable to parse", ex);
-        } catch (IOException ex) {
-            LOGGER.log(Level.WARNING, "Unable to parse", ex);
-        }
-
-        return null;
+        return new JUnitParser(archiver.isKeepLongStdio(), /* TODO really?! */false).parse(getGlob(archiver), this.owner, null, null);
```

Comment:
I see no problem with empty test results here. Maybe `true` is safer if archiver does not propagate the option

Reply:
I imagine `true` is correct, but the original code was calling a deprecated overload which effectively passed `false`, and I am leaving that untouched so as not to affect behavior for freestyle projects.